File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        StringBuilder res = new StringBuilder();

        for (char ch : s.toCharArray()) {
            if (ch >= 'a' && ch <= 'z') {
                res.append(ch);
            } else if (ch == '*') {
                if (res.length() > 0)
                    res.deleteCharAt(res.length() - 1);
            } else if (ch == '#') {
                if (res.length() > 0)
                    res.append(res);
            } else if (ch == '%') {
                if (res.length() > 0)
                    res.reverse();
            }
        }

        return res.toString();
    }
}

File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) 
    {
        string res = "";
        for (char ch : s) 
        {
            if (ch >= 'a' && ch <= 'z') {
                res.push_back(ch);
            }
            else if (ch == '*') {
                if (!res.empty())
                    res.pop_back();
            }
            else if (ch == '#') {
                if (!res.empty())
                    res += res;
            }
            else if (ch == '%') {
                if (!res.empty())
                    reverse(res.begin(), res.end());
            }
        }
        return res;
    }
};